raiden/blockchain/events.py

```diff
@@ -329,6 +332,7 @@ def add_event_listener(self, event_name, eth_filter, abi):
             event_name,
             eth_filter,
             abi,
+            True,
```

Comment:
So wait .. let me see if I follow. You are making sure that `first_run` applies to all filters like this. And from the loop above if it's no longer the first run you set the flag to false.

Reply:
yep, I've added this `first_run` flag to the `EventListener` instead of having one in `BlockchainEvents` - we need to call `get_all_entries` also for filters created on-fly (i.e. new channel)